services/autoscaling/tests/unit/test_dynamic_scaling_core.py

```diff
@@ -160,11 +160,19 @@ def mock_machines_buffer(monkeypatch: pytest.MonkeyPatch) -> Iterator[int]:
     yield num_machines_in_buffer
 
 
+@pytest.fixture
+def with_valid_time_before_termination(
+    monkeypatch: pytest.MonkeyPatch,
+) -> datetime.timedelta:
+    time = "00:11:00"
+    monkeypatch.setenv("EC2_INSTANCES_TIME_BEFORE_TERMINATION", time)
```

Comment:
hehe this fixture answers a question I had above:-)

Reply:
Q: how does it work if I want to set "15 days" ? Should I use hours?

Reply:
RTFM @pcrespov :-) 
![image](https://user-images.githubusercontent.com/32402063/225031349-7f10791e-87cd-4ed9-9eed- a7cfc12001ef.png)
https://docs.pydantic.dev/usage/types/
All clear :-)

Reply:
MINOR: make sure it is positive timedelta!

Reply:
TIP: i am using this...
```python

    @validator("STUDIES_GUEST_ACCOUNT_LIFETIME")
    @classmethod
    def is_positive_lifetime(cls, v):
        if v and isinstance(v, timedelta):
            if v.total_seconds() < 0:
                raise ValueError("Must be a positive lifetime")
        return v

```

Reply:
yes there is a validator